src/pybind/cephfs/cephfs.pyx

```diff
@@ -1541,3 +1589,172 @@ cdef class LibCephFS(object):
+        """
+        Set access and modification time for path
+
+        :param path: file path for which timestamps have to be changed
+        :param times: if times is not None, it must be a tuple (atime, mtime)
+        """
+
+        self.require_state("mounted")
+        path = cstr(path, 'path')
+        if times:
+            if not isinstance(times, tuple):
+                raise TypeError('times must be a tuple')
+            if not isinstance(times[0], (int, float)):
+                raise TypeError('atime should be an int')
+            if not isinstance(times[1], (int, float)):
+                raise TypeError('mtime should be an int')
+        actime = modtime = int(time.time())
+        if times:
+            actime = int(times[0])
+            modtime = int(times[1])
+
+        cdef:
+            char *pth = path
+            utimbuf buf = utimbuf(actime, modtime)
+        with nogil:
+            ret = ceph_utime(self.cluster, pth, &buf)
+        if ret < 0:
+            raise make_ex(ret, "error in utime {}".format(path.decode('utf-8')))
+
+    def futime(self, fd, times=None):
+        """
+        Set access and modification time for a file pointed by descriptor
+
+        :param fd: file descriptor of the open file
+        :param times: if times is not None, it must be a tuple (atime, mtime)
+        """
+
+        self.require_state("mounted")
+        if not isinstance(fd, int):
+            raise TypeError('fd must be an int')
+        if times:
+            if not isinstance(times, tuple):
+                raise TypeError('times must be a tuple')
+            if not isinstance(times[0], (int, float)):
+                raise TypeError('atime should be an int')
+            if not isinstance(times[1], (int, float)):
+                raise TypeError('mtime should be an int')
+        actime = modtime = int(time.time())
+        if times:
+            actime = int(times[0])
+            modtime = int(times[1])
+
+        cdef:
+            int _fd = fd
+            utimbuf buf = utimbuf(actime, modtime)
+        with nogil:
+            ret = ceph_futime(self.cluster, _fd, &buf)
+        if ret < 0:
+            raise make_ex(ret, "error in utime")
+
+    def utimes(self, path, times=None, follow_symlink=True):
+        """
+        Set access and modification time for path
+
+        :param path: file path for which timestamps have to be changed
+        :param times: if times is not None, it must be a tuple (atime, mtime)
+        :param follow_symlink: perform the operation on the target file if @path
+                               is a symbolic link (default)
+        """
+
+        self.require_state("mounted")
+        path = cstr(path, 'path')
+        if times:
+            if not isinstance(times, tuple):
+                raise TypeError('times must be a tuple')
+            if not isinstance(times[0], (int, float)):
+                raise TypeError('atime should be an float')
```

Comment:
I'm confused. Why do you check whether it's of type int or float? Checking for float should be good enough, right, as is in the error message?

Reply:
one could pass in a time value with seconds granularity too, i.e.,  tv_usec (in struct timeval) can be 0